src/systems/world_exporter/WorldExporter.cc

```diff
@@ -0,0 +1,382 @@
+
+          igndbg << "Uri : " << _geom->Data().MeshShape()->Uri() << std::endl;
+          igndbg << "Path : " << _geom->Data().MeshShape()->FilePath()
+            << std::endl;
+          igndbg << "fullPath : " << fullPath << std::endl;
+
+          if (fullPath.empty())
+          {
+            ignerr << "Mesh geometry missing uri" << std::endl;
+            return true;
+          }
+          mesh = meshManager->Load(fullPath);
+
+          igndbg << "Submesh count : " << mesh->SubMeshCount() << std::endl;
+
+          if (!mesh) {
+            ignerr << "MESH NOT FOUND!" << std::endl;
+            return true;
+          }
+
+          for (unsigned int k = 0; k < mesh->SubMeshCount(); k++)
+          {
+            igndbg << ">>>>>>>>> SUBMESH : " << k << std::endl;
+            int j = mesh->SubMeshByIndex(k).lock()->MaterialIndex();
+
+            igndbg << "Query Material index : " << j << std::endl;
+
+            igndbg << "Material : " << mesh->MaterialByIndex(
+              mesh->SubMeshByIndex(k).lock()->MaterialIndex()) << std::endl;
+            igndbg << "Material Ptr: " << mesh->MaterialByIndex(
+              mesh->SubMeshByIndex(k).lock()->MaterialIndex()).get()
+              << std::endl;
+
+            igndbg << "Material index on worldmesh: " <<
+              this->worldMesh.IndexOfMaterial(mesh->MaterialByIndex(
+                mesh->SubMeshByIndex(k).lock()->MaterialIndex()).get())
+                << std::endl;
+
+            int i = 0;
+            if ( j != -1 )
+            {
+              i = this->worldMesh.IndexOfMaterial(mesh->MaterialByIndex(
+                mesh->SubMeshByIndex(k).lock()->MaterialIndex()).get());
+              if (i < 0)
+              {
+                i = this->worldMesh.AddMaterial(mesh->MaterialByIndex(
+                  mesh->SubMeshByIndex(k).lock()->MaterialIndex() ));
+              }
+            }
+            else
+            {
+              i = this->worldMesh.AddMaterial(mat);
+            }
+
+            igndbg << "Inserted Material index : " << i << std::endl;
+
+            igndbg << "Texture Image : "
+              << this->worldMesh.MaterialByIndex(i)->TextureImage()
+              << std::endl;
+
+            subm = this->worldMesh.AddSubMesh(
+              *mesh->SubMeshByIndex(k).lock().get());
+            subm.lock()->SetMaterialIndex(i);
+            igndbg << "Scale : " << _geom->Data().MeshShape()->Scale()
+              << std::endl;
+
+            subm.lock()->Scale(_geom->Data().MeshShape()->Scale());
+            subMeshMatrix.push_back(matrix);
+
+            igndbg << ">>>>>>>>>" << std::endl;
+          }
+      }
+      else
+      {
+        ignerr << "It is NOT a mesh!"  << std::endl;
+      }
+
+      return true;
+    });
+
+    exporter.Export(&this->worldMesh, "./" + this->worldMesh.Name(), true,
+                    subMeshMatrix);
+    this->rendered = true;
+  }
+};
+
+WorldExporter::WorldExporter() : System(),
```

Comment:
Style nitpick, all functions should have 50 `/` before them:

```suggestion
//////////////////////////////////////////////////
WorldExporter::WorldExporter() : System(),
```